### DISCUSSION TITLE
how to cleanup the old data from the airflow DB (cleaning up the disk space )

### DISCUSSION THREAD

**Discussion by vghar-bh** [apache/airflow#34467] [Q&A] [answered]
I am currently using Airflow 2.4.1 with the Celery Executor.
I used airflow db clean --clean-before-timestamp '2023-07-01 00:00:00+00' -t tablename comand to Purge old records in metastore tables .
when i ran the command in one of my airflow pod the output was showing the old data was moved to airflow_delete__tablename and then i dropped the  airflow_delete__tablename.
well i tried to view the old data in the table it was not available but the disk space was not cleaned it was showing the same which was there before cleanup.
and in dev-airflow-postgresql-0 pod at this path [ @dev-airflow-postgresql-0:/bitnami/postgresql/data] i see the disk space is still full.
**output**: I have no name!@dev-airflow-postgresql-0:/bitnami/postgresql/data$ du -h
7.4M    ./base/1
12K     ./base/pgsql_tmp
11G     ./base/16384
7.4M    ./base/13103
7.6M    ./base/13104
11G     ./base

- storage size of airflow PVC is 12 GB
- airflow version 2.4.1

![airflow db clean output](https://github.com/apache/airflow/assets/109506747/f3883a12-1017-40ce-9b57-1a09650b58e6)

Any advice or insights from the community would be greatly appreciated!

Thank you,

**Comment by Taragolis** [comment, marked as answer]
Welcome to world of WAL based RDBMS. Short explanation let's imagine you write 10 GB of the data into the table, and after that you delete 5GB of the data, what actually happen internally: 
- Postgres marks rows (tuples) as deleted
- autovacuum daemon mark it as available to rewrite but at your disk table file still use 10GB.

You need to run [VACUUM FULL](https://www.postgresql.org/docs/current/sql-vacuum.html) against the table but be cautious:
- It will lock table until process completed
- It required additional space

**Reply by vghar-bh** [reply]
@Taragolis  sure will try it out and update 
thank you for the info
